app/src/main/java/me/eugeniomarletti/renderthread/RenderThreadDelegate.java

```diff
@@ -0,0 +1,334 @@
+
+        if (property.isHardware()) {
+            return createHardwarePaintAlphaAnimator(canvas, (HardwareCanvasProperty<Paint>) property, targetValue);
+        } else {
+            return createSoftwarePaintAlphaAnimator(drawable, (SoftwareCanvasProperty<Paint>) property, Math.round(targetValue));
+        }
+    }
+
+    @NonNull
+    public Animator createPaintStrokeWidthAnimator(
+            @Nullable Drawable drawable,
+            @NonNull Canvas canvas,
+            @NonNull CanvasProperty<Paint> property,
+            float targetValue) {
+
+        if (property.isHardware()) {
+            return createHardwarePaintStrokeWidthAnimator(canvas, (HardwareCanvasProperty<Paint>) property, targetValue);
+        } else {
+            return createSoftwarePaintStrokeWidthAnimator(drawable, (SoftwareCanvasProperty<Paint>) property, targetValue);
+        }
+    }
+
+    @NonNull
+    protected Animator createHardwareFloatAnimator(
+            @DisplayListCanvas @Nullable Canvas canvas, @NonNull HardwareCanvasProperty<Float> property, float targetValue) {
+
+        throw new UnsupportedOperationException();
+    }
+
+    @NonNull
+    protected Animator createHardwarePaintAlphaAnimator(
+            @DisplayListCanvas @Nullable Canvas canvas,
+            @NonNull HardwareCanvasProperty<Paint> property,
+            @FloatRange(from = 0f, to = 255f) float targetValue) {
+
+        throw new UnsupportedOperationException();
+    }
+
+    @NonNull
+    protected Animator createHardwarePaintStrokeWidthAnimator(
+            @DisplayListCanvas @Nullable Canvas canvas, @NonNull HardwareCanvasProperty<Paint> property, float targetValue) {
+
+        throw new UnsupportedOperationException();
+    }
+
+    @NonNull
+    protected Animator createSoftwareFloatAnimator(
+            @Nullable final Drawable drawable, @NonNull final SoftwareCanvasProperty<Float> property, float targetValue) {
+
+        ValueAnimator animator = ValueAnimator.ofFloat(property.getValue(), targetValue);
+        animator.addUpdateListener(
+                new ValueAnimator.AnimatorUpdateListener() {
+                    @Override
+                    public void onAnimationUpdate(ValueAnimator animation) {
+                        property.setValue((float) animation.getAnimatedValue());
+                        if (drawable != null) {
+                            drawable.invalidateSelf();
+                        }
+                    }
+                });
+        return animator;
+    }
+
+    @NonNull
+    protected Animator createSoftwarePaintAlphaAnimator(
+            @Nullable final Drawable drawable, @NonNull final SoftwareCanvasProperty<Paint> property, @IntRange(from = 0, to = 255) int targetValue) {
+
+        ValueAnimator animator = ValueAnimator.ofInt(property.getValue().getAlpha(), Math.round(targetValue));
+        animator.addUpdateListener(
+                new ValueAnimator.AnimatorUpdateListener() {
```

Comment:
Can we extract the ValueAnimator in a class that we instantiate when we need it?